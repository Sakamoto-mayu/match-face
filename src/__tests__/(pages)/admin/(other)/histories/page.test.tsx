import { render,screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import HistoriesPage from '@/app/(pages)/admin/(other)/histories/page'
import React from 'react'
import "@testing-library/jest-dom"
import {AppRouterContextProviderMock} from './app-router-context-provider-mock'

describe('管理者/履歴一覧画面のテスト',() => {
    const user = userEvent.setup()
    const push = jest.fn()
    beforeEach(()=>{
        render(<AppRouterContextProviderMock router={{ push }}><HistoriesPage /></AppRouterContextProviderMock>);
    })
    const mock_answered_user = [
        {id:1,project_id:1,answer_status:true,user_name:"田中太郎"},
        {id:2,project_id:1,answer_status:false,user_name:"山田次郎"},
    ]

    describe('スナップショットテスト',() => {
        it('レンダリング時',async() => {
            const view = render(<HistoriesPage/>)
            expect(view.container).toMatchSnapshot()
        })
    })
    describe('リストテスト',()=>{
        it('オープンボタンを押すとクローズボタンとユーザーリストが表示される',async()=>{
            const openButton = screen.getByTestId('open_1');
            await user.click(openButton);
            const closeButton = screen.getByTestId('close_1');
            const userList = screen.getByText('氏名')
            expect(closeButton).toBeInTheDocument();
            expect(userList).toBeInTheDocument();
            expect(openButton).not.toBeInTheDocument();
        })
        it('詳細ボタンを押すとページ遷移する',async()=>{
            const openButton = screen.getByTestId('open_1')
            await (user.click(openButton))
            const detailButton = screen.getByTestId('detail_1')
            expect(detailButton).toBeTruthy()
            await user.click(detailButton)
            expect(push).toBeCalled()
            expect(push).toBeCalledWith('/admin/review/1')
        })
        it('状態に合わせて表示が変わる',async()=>{
            // fetchでやる
        })
    })
    describe('絞り込みテスト',()=>{
        it('回答月の選択',async()=>{
            const beforeSelectMonth = screen.getByRole("option",{name:"--"}) as HTMLOptionElement
            expect(beforeSelectMonth.selected).toBe(true)
            await user.selectOptions(screen.getByTestId('month'),'2023-07')
            const selectMonth = screen.getByRole("option",{name:"2023-07"}) as HTMLOptionElement
            expect(selectMonth.selected).toBe(true)
            expect(beforeSelectMonth.selected).toBe(false)
        })
        it('職種の選択',async()=>{
            const selectJava = screen.getByRole('button',{name:"Java"})
            await user.click(selectJava)
            expect(selectJava).toHaveClass("bg-deep-gray")
            await user.click(selectJava)
            expect(selectJava).toHaveClass("bg-white")
        })
        it('技術の選択',async()=>{
            const selectTypescript = screen.getByRole('checkbox',{name:"TypeScript"})
            const selectReact = screen.getByRole('checkbox',{name:"React"})
            // Typescriptをcheck
            await user.click(selectTypescript)
            expect(selectTypescript).toBeChecked()
            // TypescriptとReactをcheck
            await user.click(selectReact)
            expect(selectTypescript).toBeChecked()
            expect(selectReact).toBeChecked()
            // Typescriptのcheckを解除
            await user.click(selectTypescript)
            expect(selectTypescript).not.toBeChecked()
            expect(selectReact).toBeChecked()
        })
        it('絞り込みボタンを押下',async()=>{
            const consoleMock = jest.fn()
            global.console.log = consoleMock
            const selectMonth = screen.getByTestId('month')
            const selectJava = screen.getByRole('button',{name:"Java"})
            const selectTypescript = screen.getByRole('checkbox',{name:"TypeScript"})
            await user.selectOptions(selectMonth,'2023-07')
            await user.click(selectJava)
            await user.click(selectTypescript)
            const form = screen.getByRole('button',{name:"絞り込み"})
            await user.click(form)
            expect(consoleMock).toHaveBeenCalled()
            expect(consoleMock).toHaveBeenCalledTimes(1)
            expect(consoleMock).toHaveBeenCalledWith({ month: '2023-07', department: 'Java', skills: [ 'TypeScript' ] })
        })
    })
})
