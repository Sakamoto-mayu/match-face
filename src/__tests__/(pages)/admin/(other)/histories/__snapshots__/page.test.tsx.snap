// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`管理者/履歴一覧画面のテスト スナップショットテスト レンダリング時 1`] = `
<div>
  <section
    class="w-[75vw] ml-[12.5vw] border-2 text-center my-[5vh] py-[5vh]"
  >
    <form
      action="submit"
      data-testid="form"
    >
      <div
        class="mb-[2vh]"
      >
        <label
          for="month"
        >
          回答月：
        </label>
        <select
          class="border-2"
          data-testid="month"
          id="month"
          name="month"
        >
          <option
            value=""
          >
            --
          </option>
          <option
            value="2023-07"
          >
            2023-07
          </option>
          <option
            value="2023-08"
          >
            2023-08
          </option>
        </select>
      </div>
      <div
        class="mb-[2vh]"
      >
        <span>
          <button
            class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm py-2 px-4 rounded-full border shadow-md mx-[10px] w-[8vw] max-sm:w-[15vw] max-sm:mb-[8px]"
            value="1"
          >
            FR
          </button>
        </span>
        <span>
          <button
            class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm py-2 px-4 rounded-full border shadow-md mx-[10px] w-[8vw] max-sm:w-[15vw] max-sm:mb-[8px]"
            value="2"
          >
            Java
          </button>
        </span>
        <span>
          <button
            class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm py-2 px-4 rounded-full border shadow-md mx-[10px] w-[8vw] max-sm:w-[15vw] max-sm:mb-[8px]"
            value="3"
          >
            PHP
          </button>
        </span>
        <span>
          <button
            class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm py-2 px-4 rounded-full border shadow-md mx-[10px] w-[8vw] max-sm:w-[15vw] max-sm:mb-[8px]"
            value="4"
          >
            CL
          </button>
        </span>
        <span>
          <button
            class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm py-2 px-4 rounded-full border shadow-md mx-[10px] w-[8vw] max-sm:w-[15vw] max-sm:mb-[8px]"
            value="5"
          >
            ML
          </button>
        </span>
        <span>
          <button
            class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm py-2 px-4 rounded-full border shadow-md mx-[10px] w-[8vw] max-sm:w-[15vw] max-sm:mb-[8px]"
            value="6"
          >
            QA
          </button>
        </span>
      </div>
      <div
        class="lg:flex justify-center mb-[2vh] max-w-[55vw] ml-[10vw]"
      >
        <legend
          class="lg:w-[215px] lg:h-10 lg:leading-10"
        >
          使用技術：
        </legend>
        <fieldset
          class="text-justify"
          id="skill"
          name="skill"
        >
          <span
            class="px-[10px] whitespace-nowrap"
          >
            <input
              id="JavaScript"
              name="JavaScript"
              type="checkbox"
              value="1"
            />
            <label
              for="JavaScript"
            >
              JavaScript
            </label>
          </span>
          <span
            class="px-[10px] whitespace-nowrap"
          >
            <input
              id="TypeScript"
              name="TypeScript"
              type="checkbox"
              value="2"
            />
            <label
              for="TypeScript"
            >
              TypeScript
            </label>
          </span>
          <span
            class="px-[10px] whitespace-nowrap"
          >
            <input
              id="React"
              name="React"
              type="checkbox"
              value="3"
            />
            <label
              for="React"
            >
              React
            </label>
          </span>
          <span
            class="px-[10px] whitespace-nowrap"
          >
            <input
              id="Next.js"
              name="Next.js"
              type="checkbox"
              value="4"
            />
            <label
              for="Next.js"
            >
              Next.js
            </label>
          </span>
          <span
            class="px-[10px] whitespace-nowrap"
          >
            <input
              id="Vue.js"
              name="Vue.js"
              type="checkbox"
              value="5"
            />
            <label
              for="Vue.js"
            >
              Vue.js
            </label>
          </span>
          <span
            class="px-[10px] whitespace-nowrap"
          >
            <input
              id="Angular"
              name="Angular"
              type="checkbox"
              value="6"
            />
            <label
              for="Angular"
            >
              Angular
            </label>
          </span>
          <span
            class="px-[10px] whitespace-nowrap"
          >
            <input
              id="Nuxt.js"
              name="Nuxt.js"
              type="checkbox"
              value="7"
            />
            <label
              for="Nuxt.js"
            >
              Nuxt.js
            </label>
          </span>
          <span
            class="px-[10px] whitespace-nowrap"
          >
            <input
              id="jQuery"
              name="jQuery"
              type="checkbox"
              value="8"
            />
            <label
              for="jQuery"
            >
              jQuery
            </label>
          </span>
          <span
            class="px-[10px] whitespace-nowrap"
          >
            <input
              id="Redux"
              name="Redux"
              type="checkbox"
              value="9"
            />
            <label
              for="Redux"
            >
              Redux
            </label>
          </span>
          <span
            class="px-[10px] whitespace-nowrap"
          >
            <input
              id="vuex"
              name="vuex"
              type="checkbox"
              value="10"
            />
            <label
              for="vuex"
            >
              vuex
            </label>
          </span>
          <span
            class="px-[10px] whitespace-nowrap"
          >
            <input
              id="MaterialUI"
              name="MaterialUI"
              type="checkbox"
              value="11"
            />
            <label
              for="MaterialUI"
            >
              MaterialUI
            </label>
          </span>
          <span
            class="px-[10px] whitespace-nowrap"
          >
            <input
              id="tailwind"
              name="tailwind"
              type="checkbox"
              value="12"
            />
            <label
              for="tailwind"
            >
              tailwind
            </label>
          </span>
          <span
            class="px-[10px] whitespace-nowrap"
          >
            <input
              id="Bootstrap"
              name="Bootstrap"
              type="checkbox"
              value="13"
            />
            <label
              for="Bootstrap"
            >
              Bootstrap
            </label>
          </span>
          <span
            class="px-[10px] whitespace-nowrap"
          >
            <input
              id="Vercel"
              name="Vercel"
              type="checkbox"
              value="14"
            />
            <label
              for="Vercel"
            >
              Vercel
            </label>
          </span>
        </fieldset>
      </div>
      <div>
        <button
          class="bg-orange drop-shadow-md hover:saturate-150 active:drop-shadow-none active:shadow-inner hover:bg-depp-orange active:bg-depp-orange active:translate-y-0.5 text-white rounded-xl p-2 w-40 h-auto text-lg"
          type="submit"
        >
          絞り込み
        </button>
      </div>
    </form>
  </section>
  <section>
    <table
      class="border-collapse border border-slate-deep-gray w-[75vw] ml-[12.5vw] text-center mb-[10vh]"
    >
      <thead>
        <tr>
          <th
            class="border border-slate-deep-gray bg-light-gray w-[5vw]"
            id="open"
          />
          <th
            class="border border-slate-deep-gray bg-light-gray w-[15vw]"
            id="answer_deadline"
          >
            回答期限
          </th>
          <th
            class="border border-slate-deep-gray bg-light-gray w-[15vw]"
            id="project_name"
          >
            案件名
          </th>
          <th
            class="border border-slate-deep-gray bg-light-gray w-[40vw]"
            id="project_detail"
          >
            案件概要
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            class="border border-slate-deep-gray"
            id="open"
          >
            <button
              class="text-deep-gray"
              data-testid="open_1"
              id="open_button"
              name="open_button"
              type="button"
            >
              ▶︎
            </button>
          </td>
          <td
            class="border border-slate-deep-gray"
            id="answer_deadline"
          >
            2023-07-10
          </td>
          <td
            class="border border-slate-deep-gray"
            id="project_name"
          >
            案件A
          </td>
          <td
            class="border border-slate-deep-gray"
            data-testid="project_detail_1"
            id="project_detail"
          >
            案件Aの説明
          </td>
        </tr>
      </tbody>
      <tbody>
        <tr>
          <td
            class="border border-slate-deep-gray"
            id="open"
          >
            <button
              class="text-deep-gray"
              data-testid="open_2"
              id="open_button"
              name="open_button"
              type="button"
            >
              ▶︎
            </button>
          </td>
          <td
            class="border border-slate-deep-gray"
            id="answer_deadline"
          >
            2023-07-10
          </td>
          <td
            class="border border-slate-deep-gray"
            id="project_name"
          >
            案件B
          </td>
          <td
            class="border border-slate-deep-gray"
            data-testid="project_detail_2"
            id="project_detail"
          >
            案件Bの説明案件Bの説明案件Bの説明案件Bの説明案件Bの説明案件Bの説
            ...
          </td>
        </tr>
      </tbody>
      <tbody>
        <tr>
          <td
            class="border border-slate-deep-gray"
            id="open"
          >
            <button
              class="text-deep-gray"
              data-testid="open_3"
              id="open_button"
              name="open_button"
              type="button"
            >
              ▶︎
            </button>
          </td>
          <td
            class="border border-slate-deep-gray"
            id="answer_deadline"
          >
            2023-07-15
          </td>
          <td
            class="border border-slate-deep-gray"
            id="project_name"
          >
            案件C
          </td>
          <td
            class="border border-slate-deep-gray"
            data-testid="project_detail_3"
            id="project_detail"
          >
            案件Cの説明
          </td>
        </tr>
      </tbody>
      <tbody>
        <tr>
          <td
            class="border border-slate-deep-gray"
            id="open"
          >
            <button
              class="text-deep-gray"
              data-testid="open_4"
              id="open_button"
              name="open_button"
              type="button"
            >
              ▶︎
            </button>
          </td>
          <td
            class="border border-slate-deep-gray"
            id="answer_deadline"
          >
            2023-07-15
          </td>
          <td
            class="border border-slate-deep-gray"
            id="project_name"
          >
            案件D
          </td>
          <td
            class="border border-slate-deep-gray"
            data-testid="project_detail_4"
            id="project_detail"
          >
            案件Dの説明
          </td>
        </tr>
      </tbody>
      <tbody>
        <tr>
          <td
            class="border border-slate-deep-gray"
            id="open"
          >
            <button
              class="text-deep-gray"
              data-testid="open_5"
              id="open_button"
              name="open_button"
              type="button"
            >
              ▶︎
            </button>
          </td>
          <td
            class="border border-slate-deep-gray"
            id="answer_deadline"
          >
            2023-08-10
          </td>
          <td
            class="border border-slate-deep-gray"
            id="project_name"
          >
            案件E
          </td>
          <td
            class="border border-slate-deep-gray"
            data-testid="project_detail_5"
            id="project_detail"
          >
            案件Eの説明
          </td>
        </tr>
      </tbody>
    </table>
  </section>
</div>
`;
